jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/BuildImageStep.java

```diff
@@ -77,19 +77,12 @@
 
   @Override
   public AsyncStep<Image<Layer>> call() throws ExecutionException {
-    List<ListenableFuture<?>> dependencies = new ArrayList<>();
-
-    for (PullAndCacheBaseImageLayerStep pullAndCacheBaseImageLayerStep :
-        NonBlockingSteps.get(pullAndCacheBaseImageLayersStep)) {
-      dependencies.add(pullAndCacheBaseImageLayerStep.getFuture());
-    }
-    for (BuildAndCacheApplicationLayerStep buildAndCacheApplicationLayerStep :
-        buildAndCacheApplicationLayerSteps) {
-      dependencies.add(buildAndCacheApplicationLayerStep.getFuture());
-    }
+    AsyncDependencies dependencies =
+        AsyncDependencies.using(listeningExecutorService)
+            .addSteps(NonBlockingSteps.get(pullAndCacheBaseImageLayersStep));
+    buildAndCacheApplicationLayerSteps.forEach(dependencies::addStep);
```

Comment:
Instead of `forEach`, can this be chained with `.addSteps(buildAndCacheApplicationLayerSteps)`?